addons/udes_routing/models/stock_rule.py

```diff
@@ -0,0 +1,46 @@
+# -*- coding: utf-8 -*-
+
+from odoo import api, fields, models, _
+
+
+class StockRule(models.Model):
+    _inherit = "stock.rule"
+
+    u_push_on_drop = fields.Boolean(
+        "Push on Drop-off",
+        default=False,
+        help="When stock is dropped in the "
+        "from_location of this path, "
+        "automatically push it onwards.",
+    )
+
+    @api.model
+    def get_path_from_location(self, location):
+        """
+        Find a single stock.rule for which the given location is
+        a valid starting location. If one not found return an empty recordset.
+        """
+        push_steps = self.search(
+            [
+                (
+                    "location_src_id",
+                    "parent_of",
+                    [
+                        location.id,
+                    ],
+                ),
+                ("u_push_on_drop", "=", True),
+            ]
+        )
+        if push_steps:
+            return push_steps[0]
+        return self.browse()
```

Comment:
If only expect one can just return the record set, if there are more then not sure which we choose?